numbert/reranker.py

```diff
@@ -0,0 +1,237 @@
+        Path to PyTorch model directory
+    model_type : str
+        Type of pretrained Transformer model used
+    max_seq_len : int
+        Maximum length of input sequence fed into the model
+    batch_size : int
+        Batch size used while evaluating re-ranker
+    '''
+    def __init__(self, index_dir, model_dir = None, model_type="bert", max_seq_len = 512, batch_size = 1):
+        super().__init__(index_dir)
+        self.model, self.model_tokenizer, self.device = load_model(model_dir)
+        self.processor = processors["msmarco"]()
+        self.output_mode = output_modes["msmarco"]
+        self.model_type = model_type
+        self.max_seq_len = max_seq_len
+        self.batch_size = batch_size
+
+    def rerank(self, q, k1=100, k2=10, t=-1):
+        '''
+        Parameters
+        ----------
+        q : str
+            Query string
+        k1 : int
+            Number of hits to return from pyserini's SimpleSearcher
+        k2 : int
+            Number of hits to return from MonoBERT re-ranker
+        t : int
+            Query tweet time for searching tweets  
+            
+        Returns
+        -------
+        results : list of Hits
+            List of document hits returned from search
+        '''
+        return self.batch_rerank(queries=[q], qids=['0'], 
+                                    k1=k1, k2=k2, t=t, threads=1)['0']
+
+    def batch_rerank(self, queries, qids, k1=100, k2=10, t=-1, threads=1):
+        '''
+        Parameters
+        ----------
+        queries : list of str
+            list of query strings
+        qids : list of str
+            list of corresponding query ids
+        k1 : int
+            Number of hits to return from pyserini's SimpleSearcher
+        k2 : int
+            Number of hits to return from MonoBERT re-ranker
+        t : int
+            Query tweet time for searching tweets  
+        threads : int
+            Maximum number of threads 
+            
+        Returns
+        -------
+        result_dict : dict of {str : Hits}
+            Dictionary of {qid : document hits} returned from each query
+        '''
+        batch_hits = super().batch_rerank(queries=queries, qids=qids, k=k1,
+                                             t=t, threads=1)
+        query_dict = dict(zip(qids, queries))
+        examples, docid_dict = self.processor.get_examples_online(query_dict, batch_hits)
+        features = convert_examples_to_features(examples,
+                                                self.model_tokenizer,
+                                                label_list=self.processor.get_labels(),
+                                                max_length=self.max_seq_len,
+                                                pad_on_left=bool(self.model_type in ['xlnet']),                 # pad on the left for xlnet
+                                                pad_token=self.model_tokenizer.convert_tokens_to_ids([self.model_tokenizer.pad_token])[0],
```

Comment:
I think the identation is wrong here, right?

Reply:
Yep, thanks for letting me know. Apparently a lot of my code's indentation/line length didn't conform to the pep8 standards (Which I'm very new to). I updated it to hopefully the right format.